### DISCUSSION TITLE
Mønstre for navigasjon i Jøkul-løsninger

### DISCUSSION THREAD

**Discussion by Stakset** [fremtind/jokul#3049] [Annet]
Jeg finner ikke menu item komponenten i Figma lengre. Er det en grunn til at denne har blitt tatt bort?

**Comment by wkillerud** [comment]
Husker jeg rett om det er den som likner på toppmenyen blant annet på https://jokul.fremtind.no @Stakset? 

Mulig den ble hoppet over i nyeste biblioteket fordi den ikke finnes i kode "offisielt" fra Jøkul. Kan hende vi skal rette på det ved å gjøre MenuItem tilgjengelig som en komponent _de jure_, ikke bare _de facto_? 😄

Cc @onordbo

**Reply by onordbo** [reply]
Det stemmer som @wkillerud sier at den ikke ligger i Figma-biblioteket ettersom det ikke finnes en kode-ekvivalent "offisielt" i designsystemet (ref: [komponentlisten](https://jokul.fremtind.no/komponenter/accordion) _fig.1_). Når det er sagt så er det ingenting i veien for å tilby det som en egen komponent.

_Fig. 1: Komponentlisten til venstre._
<img width="1178" alt="image" src="https://user-images.githubusercontent.com/4263132/188660970-3798a590-9f1c-4e24-baeb-7fe791128208.png">

**Reply by onordbo** [reply]
cc @piofinn

**Comment by ghost** [comment]
Me har ein lignande i SeOpp: https://meldskade-develop.test.fremtind.no/
Kunne gjerne tatt i bruk ein komponent frå Jøkul her.

![image](https://user-images.githubusercontent.com/89779105/188609610-b5a84473-c29b-46a3-b8a2-eb990069c751.png)

**Comment by piofinn** [comment]
Kan vi i så fall kanskje tilby et `Menu`-mønster? Det føles litt rart å bare ha en `MenuItem` for seg selv. Det sier heller ikke så mye om hvordan den brukes riktig når den er frittstående.

Ser for meg at `Menu`-mønstret/komponenten styrer størrelse/høyde, og avstand mellom menypunktene, så sørger `MenuItem`-komponenten for selve stylingen av og oppførselen til selve menypunktene.

**Reply by onordbo** [reply]
Makes sense. Spant opp en kjapp skisse på hvordan det kan funke basert på menyen vi har i portalen. Vi vil få en utfordring med antall menyvalg da man ikke kan utvide antall elementer i en instance. Alternativt kan vi duplisere ut overdrevent mange elementer og skjule de ut av boksen, så får man som bruker selv slå på ekstra menyvalg hvis man har behov for _veldig mange._

✏️ **Skisse:** https://www.figma.com/file/nGpYlGGJBZGbyJHl7A0SMu/feat%3A-menu?node-id=1035%3A8630

cc @Stakset

**Reply by onordbo** [reply]
Vi får også en spicy utfordring med å løse dette for mobil/smale skjermer, men det er kanskje nok i første omgang med en variant man kan bruke på brede skjermer? 🤷‍♂️

**Comment by piofinn** [comment]
@stakset, @onordbo, @wkillerud har dere noe imot at jeg døper om denne diskusjonen? Tenkte den kunne være et godt utgangspunkt for "Mønstre for navigasjon i Jøkul-løsninger" som vi diskuterte på forum.

**Reply by onordbo** [reply]
Kjør på @piofinn

**Comment by Stakset** [comment]
Skal vi kaste oss rundt med en arbeidsgruppe på det her? :) Så klarer vi kanskje å ligge i forkant av PM og BM sine løsninger/behov.

**Reply by piofinn** [reply]
Har dere kommet i gang med dette, @Stakset? Vi har begynt å se på dokumentasjon av mønstre for den nye portalen, og har ikke klart å holde oss helt unna å se på navigasjon 😅 @onordbo og jeg har samlet eksempler fra løsningene våre, og har noen idéer til hvordan man kanskje kan angripe dette på en måte som gjør det mulig å finne ganske universelle løsninger **og** få laget gode eksempler 😊

Vil gjerne sparre litt med dere som jobber med det her!

**Reply by Stakset** [reply]
Så spennende :) Vi har ikke komme i gang med en arbeidsgruppe akkurat. BM tester en navigasjon som ligger nært det vi tenker å bruke. @henrikhermansen har startet på utvikling av navbar for PM. 

Kanskje vi kunne tatt et møte neste uke:)?

**Reply by piofinn** [reply]
Vi skal jobbe videre med dette onsdag og torsdag, så kan gjerne ta en prat da ja 😊

**Reply by Stakset** [reply]
Jeg har høstferie de dagene, men vi tar opp tråden ved første mulighet :)
